libs/wingsdk/src/core/tree.ts

```diff
@@ -160,12 +192,27 @@ function synthDisplay(construct: IConstruct): DisplayInfo | undefined {
     return;
   }
   const display = Node.of(construct);
-  if (display.description || display.title || display.hidden) {
+
+  // generate ui data only based on direct children
+  const uiComponents: UIComponent[] = [];
+  if (
+    App.of(construct)._target === "sim" &&
```

Comment:
Why is this condition needed?

Reply:
Mmm-  I was thinking of skipping these for non-sim targets since Wing Console doesn't work on production apps today, so it would just be creating lots of extra resources. But I suppose perhaps it's a preoptimization - I'll take it out the condition for now